docs/what.md

```diff
@@ -0,0 +1,16 @@
+# All about controllers
+
+If you've come here from the [README](../README.md), you already know that this repo holds controllers. But what even _is_ a controller?
+
+The short answer is that it's our solution to managing and disseminating state changes within our products.
+
+Every application needs to manage state. Some state is temporary. Perhaps the
+user is on a screen which keeps track of whether a button is clickable/tappable depending on whether a request is taking place; but as soon as the user leaves that screen, that information is discarded. But some state is more permanent. Perhaps a screen needs to reuse information that was gathered on a previous screen, in which case the state is cached in memory; or perhaps the entire application needs to load all of the user's data when it starts, in which case the state is persisted to disk. Either way, we need a place to store that state, and that's where controllers come into play.
+
+Currently, there are two styles of creating controllers which we've dubbed [BaseController v1](../packages/base-controller/src/BaseController.ts) and [BaseController v2](../packages/base-controller/src/BaseControllerV2.ts). Both versions offer the same set of basic functionality:
+
+- The developer can preload the state (say, from a persistent location) when initializing.
+- The application can update the state at any time.
+- In another location, the application can listen for and respond to state changes (or stop listening altogether).
```

Comment:
Not sure about the use of "location" here. The first time it seems like it means a location on disk, and the second it seems like you mean somewhere else in the codebase.

Maybe something like this would be clearer:

```suggestion
- State can be set as part of initialization.
- State can be updated at any time.
- Controllers allow subscribing to state updates.
```

Reply:
Ah, gotcha, good point. 

I was trying to make it a bit more visualizable rather than abstract, and I feel like using the active form achieves that more than passive, but your rewording is more concise, so I like that. I'll workshop this.